src/porepy/models/momentum_balance.py

```diff
@@ -841,21 +845,41 @@ def bc_type_mechanics(self, sd: pp.Grid) -> pp.BoundaryConditionVectorial:
         bc.internal_to_dirichlet(sd)
         return bc
 
-    def bc_values_mechanics(self, subdomains: list[pp.Grid]) -> pp.ad.DenseArray:
+    def bc_values_mechanics(self, subdomains: Sequence[pp.Grid]) -> pp.ad.Operator:
         """Boundary values for the momentum balance.
 
         Parameters:
             subdomains: List of subdomains.
 
         Returns:
-            bc_values: Array of boundary condition values, zero by default. If combined
-            with transient problems in e.g. Biot, this should be a
-            :class:`pp.ad.TimeDependentArray` (or a variable following BoundaryGrid
-            extension).
+            bc_values: Operator of boundary condition values, which is defined on a
+            `BoundaryGrid`. Values zero by default. If combined
+            with transient problems in e.g. Biot, the values should be updated in
+            `update_all_boundary_conditions`.
 
         """
-        num_faces = sum([sd.num_faces for sd in subdomains])
-        return pp.wrap_as_ad_array(0, num_faces * self.nd, "bc_vals_mechanics")
+        boundary_grids = self.subdomains_to_boundary_grids(subdomains)
+        boundary_projection = pp.ad.BoundaryProjection(
+            self.mdg, subdomains=subdomains, dim=self.nd
+        )
+        # TODO: Neumann?
```

Comment:
I think that should be included by adding self.stress(boundary_grids) and extending said method.

Reply:
I agree